core/src/main/java/org/elasticsearch/index/translog/BufferingTranslogWriter.java

```diff
@@ -46,6 +46,7 @@ public BufferingTranslogWriter(ShardId shardId, long generation, ChannelReferenc
 
     @Override
     public Translog.Location add(BytesReference data) throws IOException {
+        ensureOpen();
```

Comment:
you mean removing the ensure open? +1 on that.  Sadly I see the same out of lock pattern in the Translog class as well:

```
        ensureOpen();
        ImmutableTranslogReader toClose = null;
        try (ReleasableLock lock = writeLock.acquire()) {
```